File: a054(customer_service)/Python/customer_service.py
code = {'A': 10, 'B': 11, 'C': 12, 'D': 13, 'E': 14, 'F': 15, 'G': 16, 'H': 17, 'I': 34,
        'J': 18, 'K': 19, 'L': 20, 'M': 21, 'N': 22, 'O': 35, 'P': 23, 'Q': 24, 'R': 25,
        'S': 26, 'T': 27, 'U': 28, 'V': 29, 'W': 32, 'X': 30, 'Y': 31, 'Z': 33}
while True:
    try:
        number = input()
        sum = 0
        for i in range(8):
            sum += int(number[i]) * (8 - i)
        for k, n in code.items():
            c = (10 - (sum + (n // 10) * 1 + (n % 10) * 9) % 10) % 10
            if c == int(number[8]):
                print(k, end = '')
        print()
    except:
        break

File: a054(customer_service)/C++/customer_service.cpp
#include <iostream>

int main(void) {
  std::string number;
  while (std::cin >> number) {
    int sum = 0;
    for (int i = 0; i < 8; i++) {
      sum += (int)number[i] * (8 - i);
    }
    int n = ((sum) % 10 + (int)number[8] - 10) % 10;
    switch (n) {
      case 0:
        std::cout << "BNZ" << '\n';
        break;
      case 1:
        std::cout << "AMW" << '\n';
        break;
      case 2:
        std::cout << "KLY" << '\n';
        break;
      case 3:
        std::cout << "JVX" << '\n';
        break;
      case 4:
        std::cout << "HU" << '\n';
        break;
      case 5:
        std::cout << "GT" << '\n';
        break;
      case 6:
        std::cerr << "FS" << '\n';
        break;
      case 7:
        std::cerr << "ER" << '\n';
        break;
      case 8:
        std::cout << "DOQ" << '\n';
        break;
      case 9:
        std::cout << "CIP" << '\n';
        break;
    }
  }

  return 0;
}

// 10 - number[8] = (sum % 10 + n % 10) % 10
